File: python/bc1983.py
aEs = 0
nEs = 0.0

qtdA = int(input())

for i in range(0, qtdA):
    cVal = input().split(" ")

    nMa = int(cVal[0])
    nota = float(cVal[1])

    if nota > nEs and nota >= 8:
        aEs = nMa
        nEs = nota

res = "Minimum note not reached" if nEs == 0 else aEs

print(f"{res}")

File: cpp/bc1983.cpp
#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

int main()
{
	int qtdA, nMa, aEs = 0;
	double nota, nEs = 0.0;
	
	cin >> qtdA;
	
	for(int i = 0; i < qtdA; i++)
	{
		cin >> nMa >> nota;
		
		if(nota > nEs && nota >= 8)
		{
			aEs = nMa;
			nEs = nota;
		}
	}
	
	string res = nEs == 0? "Minimum note not reached" : to_string(aEs);
	
}
